src/pem/_core.py

```diff
@@ -40,6 +40,16 @@ def as_bytes(self):
         """
         return self._pem_bytes
 
+    def __eq__(self, other):
+        return (type(self) == type(other) and
```

Comment:
I believe a wrong type should lead to NotImplemented like in https://github.com/hynek/attrs/blob/master/src/attr/_make.py#L244-L251